### PR TITLE
Precision and Recall Curve by Threshold

### PR DESCRIPTION
**Pull request by tweddielin**
- Add new component prthreshold-chart to draw the top-K percent precision and recall Curve by threshold (aka Ghani curve)
- Link the data to the graph.

### CODE REVIEW

**Comment by thcrock** [comment]
Merge conflict

**Comment by thcrock** [comment]
Codecov is right; since we're adding new backend code here, we should put in a test for the API call.

Check out this example, I added some utils to remove boilerplate for setting up API tests, so each endpoint test looks something like this: https://github.com/dssg/tyra/blob/master/tests/test_evaluations_modelprediction.py

You basically just add a json file to the proper path (`sample_json/evaluations/1/metric.json` in this case) representing expected API output, and then add the database data in the test that you expect to produce that JSON.

**Comment by tweddielin** [comment]
Should we open another issue specifically for the API tests?

File: webapp/controller.py [hunk 1]
```diff
@@ -71,6 +71,23 @@ def feature_importance(model_id=63, num=10):
         return jsonify({"sorry": "Sorry, no results! Please try again."}), 500
 
 
+@app.route('/evaluations/<int:model_id>/precision_recall_threshold', methods=['GET', 'POST'])
```
Comment (thcrock) [inline webapp/controller.py]: I wouldn't worry about this one

**Comment by thcrock** [comment]
Looks like Codecov is too shy to comment on the same PR twice, but coverage is up to 74% from the 70% in master, so that's awesome. Merging.

**Pull request merged**